packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-entity-types.tsx

```diff
@@ -0,0 +1,60 @@
+import { EntityType, VersionedUri } from "@blockprotocol/type-system-web";
+import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
+import {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useBlockProtocolAggregateEntityTypes } from "../../../../components/hooks/blockProtocolFunctions/ontology/useBlockProtocolAggregateEntityTypes";
+import { useInitTypeSystem } from "../../../../lib/use-init-type-system";
+
+type EntityTypesContextValues = {
+  types: Record<VersionedUri, EntityType> | null;
+};
+
+const EntityTypesContext = createContext<null | EntityTypesContextValues>(null);
+
+export const EntityTypesContextProvider = ({ children }: PropsWithChildren) => {
+  const typeSystemLoading = useInitTypeSystem();
+  const [entityTypes, setEntityTypes] = useState<
+    EntityTypesContextValues["types"] | null
+  >(null);
+  const { aggregateEntityTypes } = useBlockProtocolAggregateEntityTypes();
+
+  const fetch = useCallback(async () => {
```

Comment:
`fetch` is only used once – i'd put it inside the effect, and then the effect's dependencies will be clearer. It's always hard to know when an effect will re-run when it's dependency is a function.